### PR TITLE
IPluginGame::listSaves

### PR DESCRIPTION
**Pull request by Holt59**
See https://github.com/ModOrganizer2/modorganizer-uibase/pull/101

### CODE REVIEW

File: src/mainwindow.cpp [hunk 1]
```diff
@@ -4960,20 +4949,15 @@ void MainWindow::deleteSavegame_clicked()
   int count = 0;
 
   for (const QModelIndex &idx : ui->savegameList->selectionModel()->selectedIndexes()) {
-    QString name = idx.data(Qt::UserRole).toString();
+
+    auto& saveGame = m_SaveGames[idx.row()];
```
Comment (Al12rs) [inline src/mainwindow.cpp:4953]: If I'm reading this right you are sorting the vector `m_SaveGames` by filetime then adding the entries in order to the widget. That is what makes the `row()` number match the vector index. I don't really like this approach as it means that the view is directly reflecting the model but all the coupling is done by hand. If we ever want to allow sorting the with a proxy or something similar then this wouldn't work anymore.

Basically currently the code assumes that no one is going to reorder the vector and no one is going to change the view, or, in case that was to happen you would need control it so the indexes remain the same.

An alternative is to save the indexes of the vector in the `Qt::UserRole` of the row item. This way there is some decoupling and the view could be sorted by through a proxy without having to also sort the model.

File: src/organizer_en.ts [hunk 2]
```diff
@@ -110,22 +110,22 @@
         <translation type="unfinished"></translation>
     </message>
```
Comment (Al12rs) [inline src/organizer_en.ts:111]: Ah this got committed as well, do you think you can revert it to avoid annoying conflicts when pulling locally with umbrella?

File: src/mainwindow.cpp (hunk 1 above)
Comment (Holt59) [reply src/mainwindow.cpp:4953]: I agree, but that's probably better in a separate PR with proper display of saves ;)

Comment (Al12rs) [reply src/mainwindow.cpp:4953]: I concede to that

File: src/transfersavesdialog.cpp [hunk 3]
```diff
@@ -285,34 +217,13 @@ void TransferSavesDialog::refreshSaves(SaveCollection &saveCollection, QString c
...
-  log::debug("reading save games from {}", savesDir.absolutePath());
-
-  QFileInfoList files;
-  while (it.hasNext()) {
-    it.next();
-    files.append(it.fileInfo());
-  }
-  std::sort(files.begin(), files.end(), [](auto const& lhs, auto const& rhs) {
-    return lhs.fileTime(QFileDevice::FileModificationTime) < rhs.fileTime(QFileDevice::FileModificationTime);
+  auto saves = m_GamePlugin->listSaves(savedir);
```
Comment (Al12rs) [inline src/transfersavesdialog.cpp:220]: What happens here if the game doesn't implement `listSaves()`?

Comment (Holt59) [reply src/transfersavesdialog.cpp:220]: This is not a feature so the game always implement `listSaves`. If the game return an empty list, the list will be empty.

Comment (Al12rs) [reply src/transfersavesdialog.cpp:220]: It's not the what I would have thought knowing there is a game feature specific for saves. But I guess that is for the hover widget only?

**Review by Al12rs — APPROVED** [review]

File: src/transfersavesdialog.cpp (hunk 3 above)
Comment (Holt59) [reply src/transfersavesdialog.cpp:220]: The game feature is for widgets and missing assets.
